examples/sil/carbon-aware_control_unit/carbon_aware_control_unit.py

```diff
@@ -83,44 +106,66 @@ def scenario_step(self, current_time) -> None:
         Args:
             current_time: Current simulation time.
         """
-        nodes_power_mode = {}
-
+        nodes_power_mode_new = {}
+        battery_new = RemoteBattery()
         # Set the minimum SOC of the battery based on the current time
-        if current_time < 60*36:
-            self.battery.min_soc = 0.3
+        if current_time < 60*5:
+            battery_new.min_soc = 0.3
         else:
-            self.battery.min_soc = 0.6
+            battery_new.min_soc = 0.6
 
-        # Adjust the power modes of the nodes based on the current carbon intensity and battery SOC
+        # Adjust the power modes of the nodes based on the current carbon
+        # intensity and battery SOC
         if self.ci <= 200 or self.battery.soc > 0.8:
-            nodes_power_mode[self.nodes["aws"]] = "high performance"
-            nodes_power_mode[self.nodes["raspi"]] = "high performance"
+            nodes_power_mode_new[self.nodes["gcp"]] = "high performance"
+            #nodes_power_mode_new[self.nodes["raspi"]] = "high performance"
         elif self.ci >= 250 and self.battery.soc < self.battery.min_soc:
-            nodes_power_mode[self.nodes["aws"]] = "power-saving"
-            nodes_power_mode[self.nodes["raspi"]] = "power-saving"
+            nodes_power_mode_new[self.nodes["gcp"]] = "power-saving"
+            #nodes_power_mode_new[self.nodes["raspi"]] = "power-saving"
         else:
-            nodes_power_mode[self.nodes["aws"]] = "normal"
-            nodes_power_mode[self.nodes["raspi"]] = "normal"
-
-        # Send and forget
-        Thread(target=self.send_battery, args=(self.battery,)).start()
-        Thread(target=self.send_nodes_power_mode, args=(nodes_power_mode,)).start()
+            nodes_power_mode_new[self.nodes["gcp"]] = "normal"
+            #nodes_power_mode_new[self.nodes["raspi"]] = "normal"
+
+        # Send and forget if values changed
+        if (self.battery.min_soc != battery_new.min_soc or
+            self.battery.grid_charge != battery_new.grid_charge):
+            Thread(target=self.send_battery, args=(battery_new,)).start()
+            self.battery = battery_new
+
+        for node_id in self.nodes.values():
+            if (
+                not node_id in self.nodes_power_mode or
+                self.nodes_power_mode[node_id] != nodes_power_mode_new[node_id]
+            ):
+                Thread(
+                    target=self.send_node_power_mode,
+                    args=(node_id, nodes_power_mode_new[node_id])
+                ).start()
+                self.nodes_power_mode[node_id] = nodes_power_mode_new[node_id]
 
     def send_battery(self, battery: RemoteBattery) -> None:
         """Sends battery data to the VES API.
```

Comment:
this is sometimes called VESSIM API or VES API, let's maybe just go for "energy system API" consistently